modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -507,8 +509,15 @@ void emitterActorThread(std::shared_ptr<cv::gimpl::GIslandEmitter> emitter,
 
         // Try to obtain next data chunk from the source
         cv::GRunArg data;
-        if (emitter->pull(data))
+        bool result = false;
+        {
+            GAPI_ITT_AUTO_TRACE_GUARD(emitter_pull_hndl);
+            result = emitter->pull(data);
+        }
```

Comment:
I believe it is too old-fashioned to have modifiable temporary variables only for this.
What I'd do:

```c++
const bool have_data = [&]() {
    GAPI_ITT_AUTO_TRACE_GUARD(emitter_pull_hndl);
    return emitter->pull(data);
}();
if (have_data) {
    ...
}
```

or maybe a more extreme
```c++
if ([&]() {
        GAPI_ITT_AUTO_TRACE_GUARD(emitter_pull_hndl);
        return emitter->pull(data);
    }()) {
    ...
}
```

Reply:
Thanks a lot, fixed! 1st option is chosen!